ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Model.hs

```diff
@@ -362,6 +365,34 @@ isValid :: forall blk. LedgerSupportsProtocol blk
         -> Maybe Bool
 isValid = flip . getIsValid
 
+getHeaderStateHistory ::
+     forall blk. LedgerSupportsProtocol blk
+  => TopLevelConfig blk
+  -> Model blk -> HeaderStateHistory blk
+getHeaderStateHistory cfg m@Model { initLedger } =
+    HeaderStateHistory {
+        headerStateHistorySnapshots =
+            Seq.fromList
+          . map ((headerStates Map.!) . blockPoint)
+          . Fragment.toOldestFirst
+          $ volChain
+      , headerStateHistoryAnchor =
+          headerStates Map.! Fragment.anchorPoint volChain
+      }
+
+  where
+    volChain :: AnchoredFragment blk
+    volChain = volatileChain (configSecurityParam cfg) id m
+
+    headerStates :: Map (Point blk) (HeaderState blk)
+    headerStates = snd $
+        foldl'
+          (\(st, states) blk ->
+              let st' = tickThenReapply (extLedgerCfgFromTopLevel cfg) blk st
+              in (st', Map.insert (blockPoint blk) (headerState st') states))
+          (initLedger, Map.singleton GenesisPoint (headerState initLedger))
+          (Chain.toOldestFirst (currentChain m))
```

Comment:
Move something like this (`Chain -> HeaderStateHistory`?) in `Test.Util.Chain` (or `HeaderStateHistory` itself?) and share it with the ChainSyncClient test.